spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/SystemProperty.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.maven;
+
+/**
+ * Class for system properties passed from pom file.
+ * @author Vadim Shaigorodskiy
+ * @see RunMojo
+ * @see StartMojo
+ */
+public class SystemProperty {
+
+	private final static String NO_VALUE_FORMAT = "-D%s";
+	private final static String KEY_VALUE_FORMAT = NO_VALUE_FORMAT + "=%s";
+
+	private String key;
```

Comment:
Whatever we do, I see no reason to do any differently from what core maven plugins do. The surefire plugin had several rounds of trials apparently so I'd go with [whatever their last version is doing](http://maven.apache.org/components/surefire/maven-surefire-plugin/examples/system-properties.html)

Reply:
There are a lot of core maven plugins and far not all of them doing configuration like surefire plugin does. 
For example `maven-invoker-plugin` allows to [configure properties](https://maven.apache.org/plugins/maven-invoker-plugin/run-mojo.html#properties) in two ways: like a string line and like a map.

`maven-compiler-plugin` allows to configure [compiler arguments](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#compilerArgs) like I did with `<jvmOptions>`

Why it should be done exactly like in `maven-surefire-plugin`? To be clear I am not against that. I have just did it in one of the ways it was discussed in PR. If the `maven-surefire-plugin` is the source of truth then I can make it exactly like there.